moveit_ros/moveit_servo/src/servo.cpp

```diff
@@ -456,9 +464,27 @@ KinematicState Servo::getNextJointState(const ServoInput& command)
   KinematicState current_state(num_joints), target_state(num_joints);
   target_state.joint_names = joint_names;
 
-  // Copy current kinematic data from RobotState.
-  robot_state->copyJointGroupPositions(joint_model_group, current_state.positions);
-  robot_state->copyJointGroupVelocities(joint_model_group, current_state.velocities);
+  // remove commands not yet committed
+  auto cur_time = node_->now();
+  while (!committed_commands_.empty() &&
+         committed_commands_.back().time >
+             (cur_time + rclcpp::Duration::from_seconds(servo_params_.max_expected_latency)))
+  {
+    committed_commands_.pop_back();
+  }
```

Comment:
Basically this is pushing out the halt waypoint, correct? Seems brittle to do it based on time. I'd suggest to always pop one command each iteration.

```suggestion
  // remove the previous halt command
  auto cur_time = node_->now();
  if (!committed_commands_.empty())
  {
    committed_commands_.pop_back();
  }
```